subsys/caf/modules/ble_adv.c

```diff
@@ -888,7 +888,9 @@ static bool handle_ble_peer_event(const struct ble_peer_event *event)
 				    (state != STATE_OFF)) {
 					LOG_INF("Bonded peer power off, force system power down");
 					update_state(STATE_OFF);
-					force_power_down();
+					if (IS_ENABLED(CONFIG_CAF_FORCE_POWER_DOWN_EVENTS)) {
```

Comment:
This is not a proper fix. We need to add missing dependencies to `CONFIG_CAF_BLE_ADV_POWER_DOWN_ON_DISCONNECTION_REASON_0X15` in nRF Desktop.

`depends on DESKTOP_BLE_ADV(=y)` -> `depends on CAF_BLE_ADV_PM_EVENTS(=n) && CAF_BLE_ADV(=y) && CAF(=y)`
Otherwise we can have `CONFIG_CAF_BLE_ADV_POWER_DOWN_ON_DISCONNECTION_REASON_0X15` enabled without `CONFIG_CAF_FORCE_POWER_DOWN_EVENTS` (should not happen)

Reply:
Seems that we missed the fact that `default y` in overlay can be applied even if dependencies are not met.